File: trunk/AI/RAI/cpp-cgdk/model/Bonus.cpp
#include "Bonus.h"

using namespace model;

Bonus::Bonus(long long id, double width, double height, double x, double y, BonusType type)
: Unit(id, width, height, x, y, 0.0, 0.0, 0.0, 0.0), type_(type) { }

BonusType Bonus::type() const {
	return type_;
}

File: trunk/AI/RAI/java-cgdk/src/main/java/model/Bonus.java
package model;

/**
 * Класс, определяющий бонус --- неподвижный полезный объект. Содержит также все свойства юнита.
 */
public final class Bonus extends Unit {
    private final BonusType type;

    public Bonus(long id, double width, double height, double x, double y, BonusType type) {
        super(id, width, height, x, y, 0.0D, 0.0D, 0.0D, 0.0D);
        this.type = type;
    }

    /**
     * @return Возвращает тип бонуса.
     */
    public BonusType getType() {
        return type;
    }
}
